### PR TITLE
DownResponse in GSLB service

### PR DESCRIPTION
**Pull request by swathinsankaran**
This commit contains the crd and apis changes to configure the down response when a GSLB service is down.

### CODE REVIEW

File: gslb/cache/controller_obj_cache.go [hunk 1]
```diff
@@ -627,24 +629,70 @@ func ParsePoolAlgorithmSettingsFromPoolRaw(group map[string]interface{}) *gslbal
...
+		gslbutils.Warnf("couldn't parse down response type: %v", val)
+		return nil
+	}
+	downResponse := &gslbalphav1.DownResponse{}
+	downResponse.Type = responseType
+	if responseType != gslbalphav1.GSLBServiceDownResponseFallbackIP {
+		return downResponse
+	}
+
+	fallbackIP, ok := response["fallback_ip"].(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (swathinsankaran) [reply gslb/cache/controller_obj_cache.go:669]: `fallbackIP` is compulsory only when the user configures the responseType as `GSLB_SERVICE_DOWN_RESPONSE_FALLBACK_IP`

**Review by arihantg — DISMISSED** [review]

File: gslb/cache/controller_obj_cache.go [hunk 2]
```diff
@@ -627,24 +629,70 @@ func ParsePoolAlgorithmSettingsFromPoolRaw(group map[string]interface{}) *gslbal
 		hashMaskI := int32(hashMaskF)
 		consistentHashMask = &hashMaskI
 	} else {
-		gslbutils.Logf("couldn't parse hash mask: %v", group["consistent_hash_mask"])
+		gslbutils.Debugf("couldn't parse hash mask: %v", group)
```
Comment (aaha97) [inline gslb/cache/controller_obj_cache.go:632]: should this be a warning log?

Comment (swathinsankaran) [reply gslb/cache/controller_obj_cache.go:632]: `consistent_hash_mask` will be only present when the `algorithm` or `fallback_algorithm` is set as consistent hash.

Comment (swathinsankaran) [reply gslb/cache/controller_obj_cache.go]: As discussed, I have added a check whether the `algorithm` or `fallback_algorithm` is consistent hash, then only access the `consistent_hash_mask`.

**Review by aaha97 — APPROVED** [review]
LGTM

**Comment by swathinsankaran** [comment]
build amko

**Pull request merged**